frontend/helpers/path-helpers.ts

```diff
@@ -167,3 +167,28 @@ export function ifixitOriginWithSubdomain(subdomain: string) {
 export function joinPaths(...paths: string[]) {
    return paths.map((path) => path.replace(/^\/|\/$/g, '')).join('/');
 }
+
+export function getRouteData(route: string) {
+   const [baseRoute, ..._rest] = route.split('?');
+   const pathParts = baseRoute.split('/').filter((part) => part !== '');
+   const firstPathSegment = pathParts.length >= 1 ? pathParts[0] : '';
+   const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
+   const itemType = pathParts.length >= 3 ? pathParts[2] : '';
+   return {
+      firstPathSegment,
+      deviceHandle,
+      itemType,
+   };
+}
+
+export function getLocation(url: string) {
```

Comment:
I think you can replace https://github.com/iFixit/react-commerce/pull/2138/files#diff-f344de4d26499cec01f1344a51b8d041d21e27a457b17cae0113e03cb876caa4R72-R79

Reply:
I did attempt this, maybe i did something wrong, but the next hydration started complaining for some reason